help_to_heat/frontdoor/interface.py

```diff
@@ -164,6 +164,36 @@ def replace_in_session_data(self, session_data):
         return session_data
 
 
+class DuplicateReferralChecker:
+    def __init__(self, session_id):
+        self.session_id = session_id
+
+    def is_referral_a_duplicate(self):
+        session_data = api.session.get_session(self.session_id)
+        uprn = session_data.get("uprn")
+        return uprn and portal.models.Referral.objects.filter(data__uprn=uprn).exists()
+
+    def is_duplicate_referral_sent_to_same_energy_supplier(self):
+        if not self.is_referral_a_duplicate():
+            return False
```

Comment:
Should we throw here? Something's gone wrong if we're at this point and there's no duplicate

Reply:
sure